server/api/maprequest.ts

```diff
@@ -22,6 +22,8 @@ export default defineEventHandler(async (event: H3Event) => {
     data.style = "mapbox";
   }
 
+  const db = await getDatabaseConnection();
```

Comment:
I worry that multiple requests or threads are calling `refreshDatabaseConnection` and/or `setupDatabaseConnection` simultaneously, race conditions might occur on this `db` variable.  (I could be wrong; is defineEventHandler called once per app, or once per request?)

You might look into **connection pools**: 

```
import { Pool } from "pg";
const pool = new Pool(getConfig());
const db = await pool.connect();
```
Many threads/simultaneous requests can share this pool, and the `pg` package manages multiple clients and re-connections for you (I think) so you do not need to deal with reconnecting manually.  

I have not used this in the javascript ecosystem but I know it's a very good practice from my time in python. Maybe look for a quick tutorial to better understand if this is the right choice for you.

Reply:
In either case, last thing I'll say is, when the application shuts down, you may want to clean up database connection.  If using a pool, that would be `pool.end()`, or with your current code that's `db.end()`.  Does your current web framework let you hook in to run code on app shutdown?

Reply:
Thanks. I agreed with you that pooling is an improvement here. https://github.com/ConservationMetrics/map-packer/pull/41

I have never seen anyone implement cleaning up a database connection in Nuxt.js, but it's certainly a good suggestion and possible via hooking into the Nitro `close` lifecycle. I wrote a ticket for it: https://github.com/ConservationMetrics/gc-shared-resources/issues/3

Reply:
I'd assume that using the pg client's own connection pooling would (necessarily) handle clean up for you, so doing the one change may enable you to close both issues.  Have not verified.  Anyway, sounds good.